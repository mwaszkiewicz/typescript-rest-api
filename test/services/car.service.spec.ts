/* eslint-disable no-undef */
import { CarService } from '../../src/services/car.service';
import { pino } from 'pino';

describe('controller.service', () => {
  let service: CarService;

  beforeAll(async () => {
    service = new CarService(pino());
  });

  it('Create car', async () => {
    const result = await service.create({ vin: '12345', brand: 'Audi', model: 'A4' });
    expect(result).toMatchSnapshot();
  });

  it('Update car', async () => {
    const result = await service.update({ vin: '12345', brand: 'Audi', model: 'A4' });
    expect(result).toEqual(true);
  });

  it('Get all cars', async () => {
    const result = await service.getAll();
    expect(result.length).toEqual(2);
    expect(result).toMatchSnapshot();
  });

  it('Delete car by vin', async () => {
    const result = await service.delete('12345');
    expect(result).toEqual(1);
  });
});
