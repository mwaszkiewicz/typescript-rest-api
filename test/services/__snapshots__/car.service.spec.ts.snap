// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controller.service Create car 1`] = `
Object {
  "brand": "Audi",
  "model": "A4",
  "vin": "12345",
}
`;

exports[`controller.service Get all cars 1`] = `
Array [
  Object {
    "brand": "Audi",
    "model": "A4",
    "vin": "12345",
  },
  Object {
    "brand": "BMW",
    "model": "X3",
    "vin": "65432",
  },
]
`;
